.github/workflows/cmake.yml

```diff
@@ -73,3 +73,125 @@ jobs:
         # Execute tests defined by the CMake configuration.
         # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
         run: ctest -C ${{env.BUILD_TYPE}} || ctest -C ${{env.BUILD_TYPE}} --rerun-failed --output-on-failure
+
+  emscripten:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - emsdk: 3.1.40
+            emsdk-ext: bc
+            boost: 1_84_0
+          - emsdk: 3.1.61
+            emsdk-ext: a
+            boost: 1_85_0
+            test: true
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out repository code
+      - name: Cache Emscripten
+        uses: actions/cache@v4
+        with:
+          path: 'emsdk-cache'
+          key: ${{ runner.os }}-emsdk-${{ matrix.version.emsdk }}
+      - uses: mymindstorm/setup-emsdk@v14
+        name: Emscripten setup
+        with:
+          version: ${{ matrix.version.emsdk }}
+          actions-cache-folder: 'emsdk-cache'
+      - name: Build Boost
+        env:
+          BOOST_VER: ${{ matrix.version.boost }}
+        run: |
+          mkdir -p build
+          cd build
+          wget https://archives.boost.io/release/${BOOST_VER//_/.}/source/boost_${BOOST_VER}.tar.gz
+          tar -zxf boost_${BOOST_VER}.tar.gz
+          mv boost_${BOOST_VER} boost
+          cd boost
+          ./bootstrap.sh
+          ./b2 --prefix=dist \
+               --with-date_time --with-filesystem \
+               --with-system --with-iostreams \
+               --with-nowide --with-regex --with-test \
+               toolset=emscripten link=static variant=release threading=single install
+      - name: Build GMP
+        run: |
+          cd build
+          export GMP_VER=6.3.0
+          wget https://ftp.gnu.org/gnu/gmp/gmp-${GMP_VER}.tar.xz
+          tar -Jxf gmp-${GMP_VER}.tar.xz
+          mv gmp-${GMP_VER} gmp
+          cd gmp
+          emconfigure ./configure --disable-assembly --host none --enable-cxx --prefix=$(realpath dist)
+          emmake make
+          emmake make install
+      - name: Build MPFR
+        run: |
+          cd build
+          export MPFR_VER=4.2.1
+          wget https://ftp.gnu.org/gnu/mpfr/mpfr-${MPFR_VER}.tar.xz
+          tar -Jxf mpfr-${MPFR_VER}.tar.xz
+          mv mpfr-${MPFR_VER} mpfr
+          cd mpfr
+          emconfigure ./configure --host none --prefix=$(realpath dist) --with-gmp=$(realpath ../gmp/dist)
+          emmake make
+          emmake make install
+      - name: Build Ledger
+        env:
+          EXT: ${{ matrix.version.emsdk-ext }}
+        run: |
+          cd build
+          export BOOST="$(realpath boost/dist)"
+          # -lnodefs.js -lnoderawfs.js flags are for testing purposes.
+          emcmake cmake \
+            -DBUILD_LIBRARY=ON \
+            -DPRECOMPILE_SYSTEM_HH=OFF \
+            -DCMAKE_CXX_FLAGS=' -lnodefs.js -lnoderawfs.js ' \
```

Comment:
Also note, that I needed `-DCMAKE_CXX_FLAGS='-fwasm-exceptions'` in order to be able to compile ledger with emscripten.

Could be that I did not build a dedicated boost with a minimal set of libraries for ledger, but rather used all of the libraries that would compile with emscripten, without further tweaking.

Reply:
On CMake 3.30, using supplying `-DLINK_OPTIONS` gives a warning about the option not being used at all:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    LINK_OPTIONS

```

If it is so, it can explain why you are having this `Cannot read journal file` error, because, when using the building steps from the worflow file, filesystem access on *nix systems relies on the two flags `-lnodefs.js -lnoderawfs.js`. And if the two flags are not supplied due to `LINK_OPTIONS` not being used, native filesystem access will not be linked into the `ledger.js/wasm` files.

I changed it to [`CMAKE_EXE_LINKER_FLAGS`](https://cmake.org/cmake/help/latest/variable/CMAKE_EXE_LINKER_FLAGS.html) instead. Can you please test if it works for you?